Adding large GPMSA verification example

dmcdougall opened the pull request:
Replaces #607.

roystgnr commented on examples/gp/linear_verif/gpmsa_linear_verif_pdf.C:
```diff
@@ -0,0 +1,614 @@
+
+  // Read in data and store the standard deviation of the simulation
+  // data (ignored for now).
+  readData("sim_scalar.dat", "y_exp_scalar.txt", simulationScenarios, paramVecs,
```
Are these files included in the PR anywhere?

roystgnr replied:
Wait, never mind, those are the files that were merged in #580.

dmcdougall committed:
Adding large verification example

dmcdougall committed:
Removing normalisation code from readData function

Also converts some tabs to spaces.

dmcdougall committed:
Whitespace OCD commit

dmcdougall committed:
Delete superfluous code from readData function

dmcdougall committed:
Add comment explaining log prior calculation

For GPMSA, we expect our log prior and matlab's log prior to differ only
by an additive constant in log space.  I've added a comment that
explains this.

dmcdougall commented:
@roystgnr Ok I think I'm done here too.  Let me know if you have any comments.

dmcdougall commented:
@briadam This is the pull request where I destroyed all your hard work.  I'll resurrect it.

briadam commented:
Hardly destroyed... just morphed it into a necessary prerequisite test. ;-)  

Honestly, you'll probably do better to just write an inverse problem departing from scalar_pdf_large without looking at what I did as you'll likely then immediately see the issues I ran into with the initial positions and proposal covariance.  I can see having to scale the proposal covariance as we discussed today, but it's possible at this point (given all the work you've done) that if you take the initial point at the mean and proposal covariance to be the prior variance, you may fare better than I did, and it'll just work.

dmcdougall commented:
Well, it'll be a pull request in a couple of minutes and then you can tell me what to do with it :)
